tlatools/org.lamport.tlatools/src/tla2sany/modanalyzer/ParseUnit.java

```diff
@@ -264,10 +265,25 @@ public final void parseFile(Errors errors, boolean firstCall) throws AbortExcept
         * more useful output for the GUI.                                      *
         ***********************************************************************/
         if (ToolIO.getMode() == ToolIO.SYSTEM)
-        {
-			ToolIO.out.println(TLAFlightRecorder.message(String.format("%s %s", TLAConstants.LoggingAtoms.PARSING_FILE, absoluteResolvedPath)));
-        } else
-        {
+        {            
+            // If `resolver.getLibraryPath` returns `null` (and as we didn't have errors, the module
+            // really exists somewhere), then we assume this is part of a monolith file.
+            String originalFilePath = "";
+            String libraryPath = this.spec.getResolver().getLibraryPath(name + ".tla");
+            if (libraryPath != null) {
+                originalFilePath = " (" + libraryPath + ")";                
+            } else if (rootParseUnit != null && rootParseUnit.getNis() != null) {
+                File rootSourceFile = rootParseUnit.getNis().sourceFile();
+                if (rootSourceFile != null) {
+                    try {
+                        MonolithSpecExtractor.module(rootSourceFile, name);
+                        originalFilePath = " (" + rootSourceFile.getAbsolutePath() + ")";
+                    } catch (IOException e) {
```

Comment:
Please indicate if this is an exception that's not expected to happen or swallowed intentionally?  In case of the former, please add an e.printStackTrace() to the catch block.

Reply:
It's the former, I'll add, thanks o/